### PR TITLE
docs(type): update link to type styles in README

### PR DESCRIPTION
**Pull request by jendowns**
Currently this link to color tokens on the `@carbon/type` README just leads to the homepage, due to the deprecated next subdomain.

Also the use of "here" as link text is not very accessible, as someone using a screenreader who skips to the link won't have info in the link text about where the link goes.

#### Changelog

**Changed**

- update link to type styles on the Carbon website
- update link text to reference Carbon website as destination

### CODE REVIEW

File: packages/type/README.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ what we call type styles. These tokens have a variety of properties for styling
 how text is rendered on a page.
 
 You can find a full reference of the type styles that are available
-[here](https://next.carbondesignsystem.com/guidelines/typography/productive) .
+[Carbon Design System website](https://www.carbondesignsystem.com/guidelines/typography/productive) .
```
Comment (joshblack) [inline packages/type/README.md]: Should we update the sentence? Could you explain more about the accessibility shortcoming?

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @jendowns!

File: packages/type/README.md (hunk 1 above)
Comment (jendowns) [reply packages/type/README.md]: Oh I mean to update the sentence completely, to avoid having link text that just says "here"

Comment (jendowns) [reply packages/type/README.md]: Oh sorry just noticed your edit --

Here's two quick resources I found to explain this: 
- https://www.w3.org/QA/Tips/noClickHere
- https://www.w3.org/TR/WCAG10-HTML-TECHS/#link-text

In addition to the reasons about context provided in those links, if you are using a screenreader + tabbing between focusable elements you may just hear "Link Here" or "Link Click here" which isn't helpful.

**Comment by asudoh** [comment]
@joshblack Just in case you have further comments - Thanks!

**Review by joshblack — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]
👍

**Pull request merged**